mcs/class/corlib/System.Threading/Interlocked.cs

```diff
@@ -126,6 +126,10 @@ public static object Exchange (ref object location1, object value)
 		[Intrinsic]
 		public static T CompareExchange<T> (ref T location1, T value, T comparand) where T : class
 		{
+			unsafe {
+				if (Unsafe.AsPointer (ref location1) == null)
+					throw new NullReferenceException ();
```

Comment:
CoreCLR now has `Unsafe.IsNullRef` https://github.com/dotnet/coreclr/blob/master/src/System.Private.CoreLib/shared/Internal/Runtime/CompilerServices/Unsafe.cs#L415

Reply:
Looks like we don't have that change synced up yet.

Reply:
Cc @safern

Reply:
https://github.com/mono/mono/pull/17485